### PR TITLE
[WIP]Add spawnpoints

### PR DESCRIPTION
**Pull request by kinggoesgaming**
A `Spawnpoint` is a location holder for a `Player` to spawn at. 

Each `Spawnpoint` can be shared by multiple `Team`s. Think Team Deathmatch and similar gamemodes, where same spawn locations may be used by multiple teams.

It can be active or inactive, allowing controlled progression. 

Also, has different `SpawnpointType` and the default `SpawnpointType`s are located in `PlayerSpawnpointTypes`.

### CODE REVIEW

File: .travis/sapiver.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+5.0.0
```
Comment (Flibio) [inline .travis/sapiver.txt]: Why modify these?

**Comment by kinggoesgaming** [comment]
Gradle is doing them :/

File: .travis/sapiver.txt (hunk 1 above)
Comment (kinggoesgaming) [inline .travis/sapiver.txt]: Knew you gonna ask that 😑

**Comment by Flibio** [comment]
@kinggoesgaming: I thought that they only build when you run a specific Gradle command?

**Comment by kinggoesgaming** [comment]
Yea don't what's happening there... Might add a if check in the task to see the name of the project before doingbthat

**Comment by Flibio** [comment]
Ok, feel free to fix that first before this PR. It will help prevent future PR's from having the same issue. :+1:

**Comment by kinggoesgaming** [comment]
This is nearing finale

**Comment by kinggoesgaming** [comment]
This is will get a redo... closing

**Pull request closed without merging**